src/main/java/software/amazon/awssdk/crt/CRT.java

```diff
@@ -306,11 +311,10 @@ private static void extractAndLoadLibrary(String path) {
         String os = getOSIdentifier();
         if (os.equals("windows")) {
             tryDeleteOldLibrariesFromTempDir(tmpdirFile, tempSharedLibPrefix, libraryName);
+        } else {
+            // Once it is loaded successfully, we can safely delete it and it will keep working in the current process.
```

Comment:
trivial: Move comments so platform-specific comments go along with the platform-specific actions.
Also, give the reason why we're doing early-delete on !windows
```
// Unfortunately, we can't rely solely on File.deleteOnExit() to clean things up...
if (windows)
    // File.deleteOnExit() won't work on Windows, where
    // ...blah blah rest of windows specific comments
    tryDeleteOldLibrariesFromTempDir();
else
    // On other platforms, File.deleteOnExit() might not run if <INSERT DETAILS>
    // Therefore, we delete the library file immediately after it was successfully loaded.
    // This is safe to do on non-Windows platforms, everything will keep working in the current process.
    tempSharedLib.delete();
```